app/routers/notes/notes.py

```diff
@@ -0,0 +1,137 @@
+from typing import Any, Dict, List, Sequence
+
+from fastapi import APIRouter, Request, status, HTTPException
+from fastapi.param_functions import Depends
+from sqlalchemy.orm import Session
+from starlette.responses import RedirectResponse
+from starlette.templating import _TemplateResponse
+
+from app.database.schemas import NoteDB, NoteSchema
+from app.dependencies import get_db, templates
+from app.internal.notes import notes
+
+
+router = APIRouter(
+    prefix="/notes",
+    tags=["notes"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.post(
+    "/edit/{id}", status_code=status.HTTP_202_ACCEPTED, include_in_schema=False
+)
+async def redirect_update_note(
+    request: Request, id: int, session: Session = Depends(get_db)
+) -> RedirectResponse:
+    """Update a note using user-interface form."""
+    form = await request.form()
+    updated_note = NoteSchema(**dict(form))
+    await update_note(updated_note, id, session)
+    return RedirectResponse("/notes", status_code=status.HTTP_302_FOUND)
+
+
+@router.post(
+    "/delete/{id}", status_code=status.HTTP_200_OK, include_in_schema=False
+)
+async def redirect_delete_note(
+    id: int, session: Session = Depends(get_db)
+) -> RedirectResponse:
+    """Delete a note from the database using user-interface form."""
+    await delete_note(id, session)
+    return RedirectResponse("/notes", status_code=status.HTTP_302_FOUND)
+
+
+@router.post(
+    "/add", status_code=status.HTTP_201_CREATED, include_in_schema=False
+)
+async def create_note_by_form(
+    request: Request, session: Session = Depends(get_db)
+) -> RedirectResponse:
+    """Add a note using user-interface form."""
+    form = await request.form()
+    new_note = NoteSchema(**dict(form))
+    await notes.create_note(note=new_note, session=session)
+    return RedirectResponse("/notes", status_code=status.HTTP_302_FOUND)
+
+
+@router.post("/", response_model=NoteDB, status_code=status.HTTP_201_CREATED)
+async def create_new_note(
+    request: NoteSchema, session: Session = Depends(get_db)
+) -> Dict[str, Any]:
+    """Create a note in the database."""
+    new_note = NoteSchema(**dict(request))
+    return await notes.create_note(note=new_note, session=session)
+
+
+@router.delete("/{id}/", status_code=status.HTTP_200_OK)
+async def delete_note(id: int, session: Session = Depends(get_db)) -> str:
```

Comment:
pretty sure you got a warning here for using "id" as a variable name.

Reply:
Done